hermes/bin/src/runtime_extensions/hermes/sqlite/state.rs

```diff
@@ -0,0 +1,108 @@
+// cspell: words mapref
+
+//! Internal state implementation for the `SQLite` module.
+
+use std::collections::HashMap;
+
+use dashmap::{mapref::one::RefMut, DashMap};
+use once_cell::sync::Lazy;
+
+use crate::app::HermesAppName;
+
+/// Represents an individual state for a particular object.
+#[derive(Debug)]
+pub(crate) struct ResourceObjectState {
+    /// A map holding key-value pairs of an object ID and a value.
+    id_map: HashMap<u32, usize>,
+    /// The current incremental state of ID.
+    current_id: Option<u32>,
```

Comment:
What is the purpose of this id ?

Reply:
It's an ID to work with `wasmtime::component::Resource` as `Resource` seems to need an internal ID to work with went creating a new resource object. An incremental ID works pretty fine on this regardless of its caveat that I don't know.